src/clr-angular/utils/focus/key-focus/key-focus.spec.ts

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ClrKeyFocus } from './key-focus';
+import { Component, DebugElement, ViewChild, ElementRef, OnInit } from '@angular/core';
+import { ClrKeyFocusModule } from './key-focus.module';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+
+@Component({
+  template: `
+    <button class="btn" #enter>Enter focus</button>
+    <div *ngIf="open" [clrKeyFocus]="direction" [clrFocusOnLoad]="focusOnLoad">
+      <button class="btn">Button 1</button>
+      <button class="btn">Button 2</button>
+      <button class="btn">Button 3</button>
+    </div>
+    <button class="btn">Exit focus</button>
+  `,
+})
+class TestComponent implements OnInit {
+  @ViewChild('enter', { static: true, read: ElementRef })
+  enterElement: ElementRef;
+  @ViewChild(ClrKeyFocus, { static: true })
+  keyFocus: ClrKeyFocus;
+
+  open: boolean = false;
+  direction: string = 'vertical';
+  focusOnLoad = true;
+
+  ngOnInit() {
+    this.enterElement.nativeElement.focus();
+  }
+}
+
+let fixture: ComponentFixture<any>;
+let component: TestComponent;
+
+let directiveDebugElement: DebugElement;
+let clarityDirective: ClrKeyFocus;
+let clarityElement: HTMLElement;
+
+const openMenu = () => {
+  component.open = true;
+  fixture.detectChanges();
+
+  directiveDebugElement = fixture.debugElement.query(By.directive(ClrKeyFocus));
+  clarityElement = directiveDebugElement.nativeElement;
+  clarityDirective = directiveDebugElement.injector.get(ClrKeyFocus);
+};
+
+const keyPress = (key: string) => {
+  clarityElement.dispatchEvent(new KeyboardEvent('keydown', { key: key }));
+};
+
+describe('KeyFocus directive', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [ClrKeyFocusModule],
+      declarations: [TestComponent],
+    });
+    fixture = TestBed.createComponent(TestComponent);
+    fixture.detectChanges();
+
+    component = fixture.componentInstance;
+  });
+
```

Comment:
Is it possible to organize the suite into three, similar to how other components are tested? 
I'm asking b/c it would be nice if we followed this format: 
1. Typscript API - testing side efeects from subscriptions 
2. Template API - Inputs/Outputs/HostListeners and any public getters and setters
3. View Basics - correct css classes and other DOM structure and DOM element attribute testing (aria-*)

Not having this won't be a blocker for me but I thought I would ask :)

Reply:
Yes, I'll try to at least name it as closely as possible. Only the default/custom division (by test component) may be a little challenge, but I may move it inside, or transform the beforeeach to a helper method.
Make sense to follow the same format.

Reply:
done

Reply:
ty